woodwork/type_sys/type_system.py

```diff
@@ -267,25 +262,20 @@ def infer_logical_type(self, series):
             series = series.head(100000).to_pandas()
 
         def get_inference_matches(types_to_check, series, type_matches=[]):
-            # Since NaturalLanguage isn't inferred by default, make sure to check
-            # any children of NaturalLanguage, otherwise they never get evaluated
-            if NaturalLanguage in types_to_check:
-                check_next = self._get_children(NaturalLanguage)
-            else:
-                check_next = []
+            check_next = []
             for logical_type in types_to_check:
                 inference_func = self.inference_functions.get(logical_type)
                 if inference_func and inference_func(series):
                     type_matches.append(logical_type)
-                    check_next.extend(self._get_children(logical_type))
+                check_next.extend(self._get_children(logical_type))
```

Comment:
I think you only want to extend `check_next` with children in two situations:
1. There is an inference function and it returns `True` for the series
2. The inference function is `None`

You don't want to add children if there is an inference function an it returns `False` Otherwise you will be checking all children, even if the parent type failed inference. For example, with the code here if the `IntegerNullable` inference failed, you would still grab its children (`Integer`) and run that inference function. That is wasted compute since we know if the parent fails inference the child will as well.